exporter/datadogexporter/config/config.go

```diff
@@ -89,27 +93,41 @@ type TracesConfig struct {
 // It is embedded in the configuration
 type TagsConfig struct {
 	// Hostname is the host name for unified service tagging.
+	// It can also be set through the `DD_HOST` environment variable.
 	// If unset, it is determined automatically.
 	// See https://docs.datadoghq.com/agent/faq/how-datadog-agent-determines-the-hostname
 	// for more details.
 	Hostname string `mapstructure:"hostname"`
 
 	// Env is the environment for unified service tagging.
+	// It can also be set through the `DD_ENV` environment variable.
 	Env string `mapstructure:"env"`
 
 	// Service is the service for unified service tagging.
+	// It can also be set through the `DD_SERVICE` environment variable.
 	Service string `mapstructure:"service"`
 
 	// Version is the version for unified service tagging.
+	// It can also be set through the `DD_VERSION` environment variable.
 	Version string `mapstructure:"version"`
 
+	// EnvVarTags is the list of space-separated tags passed by the `DD_TAGS` environment variable
+	// Superseded by Tags if the latter is set.
+	// Should not be set in the user-provided config.
+	EnvVarTags string `mapstructure:"envvartags"`
```

Comment:
If we make this field private it's ignored by `mapstructure` and we avoid an user manually setting this field. This means adding some methods to `TagsConfig` but I think it's cleaner.

Reply:
This wouldn't work because If we make this field private, it's also ignored by the environment variable replacement logic (https://github.com/open-telemetry/opentelemetry-collector/blob/51563860af5a558b6e7087feea2502b5d348375c/config/config.go#L708), because you can't externally modify unexported private fields.

We'd need a custom `(t *TagsConfig)`method to expand this field, which reuses the env var expansion logic somehow, and call it manually at an appropriate time (probably in `CreateDefaultConfig`, but that means the expansion is not done at the same time as for the other env variables). I'm not sure which solution is cleaner here.

Reply:
> This wouldn't work because If we make this field private, it's also ignored by the environment variable replacement logic (https://github.com/open-telemetry/opentelemetry-collector/blob/51563860af5a558b6e7087feea2502b5d348375c/config/config.go#L708), because you can't externally modify unexported private fields.

Ah, hadn't thought about that. Yeah, then there's not much benefit on doing it differently.